### ISSUE TITLE
[BUG] Condition chaining not working for `java.referenced` conditions

### ISSUE THREAD

**Issue by jmle** [konveyor/analyzer-lsp#718]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Konveyor version

0.5.3

### Priority

Critical

### Current Behavior

A condition like the following:

```yaml
  when:
    or:
    - java.referenced:
        pattern: org.springframework.stereotype.Controller
        location: ANNOTATION
      as: class
      ignore: true
    - java.referenced:
        pattern: '* javax.xml.transform.Source'
        location: METHOD
      from: class
```
doesn't work because the chaining does not seem to work.

### Expected Behavior

Said condition should work. We should at least be sending the class file to the chain, and further filtering in the next condition, but that doesn't seem to be the case.

### How Reproducible

Always (Default)

### Steps To Reproduce

1. Run the above rule against the following project:

[web-applications.zip](https://github.com/user-attachments/files/17701951/web-applications.zip)

2. One and only one issue should appear.

### Environment

```markdown

```

**Comment by jmle** [comment]
Closed with https://github.com/konveyor/analyzer-lsp/pull/721
